### DISCUSSION TITLE
Walkthrough

### DISCUSSION THREAD

**Discussion by qillerr** [TriliumNext/Trilium#1418] [Q&A]
Hi, I'm trying to understand code and all. I've been going through the codebase, but it's really sphagetti and not object oriented, I don't understand it, but the fact that it's all in pretty much the same directory is not helping.
Why is search logged and it's visible, and all these kinds of design intentions.

Best regards, qillerr

**Comment by zadam** [comment]
Hi, I don't know how to answer that. I can answer some specific questions if you have some.

> Why is search logged and it's visible, and all these kinds of design intentions.

You probably checked out master which is work in progress. It would be better to use stable branch as a starting point.
